proctoru/proctoru.py

```diff
@@ -188,6 +188,96 @@ def get_block_id(self):
         """
```

Comment:
Well, that's a great question :)
Mostly is repeated code, and I have compared carefully the old code with the new one (with a diff) whenever was doing the replacement.
However, you are right in saying that there could be issues. I was also a bit nervous while doing that, thinking that it could have some side effect.
I would say that it is a good reason to make sure we have some really good testing going on before it lands in production. So yes, there is a big warning sign attached to this modif. We have to test it carefully !
Apart from that, it improves greatly the readability of those enormous if nested conditions....